src/org/zaproxy/zap/extension/jsonView/HttpPanelJsonView.java

```diff
@@ -0,0 +1,185 @@
+	 *
+	 * @see org.zaproxy.zap.view.messagecontainer.MessageContainer
+	 */
+	public static final String DEFAULT_MESSAGE_CONTAINER_NAME = "HttpMessagePanel";
+	public static final String NAME = "HttpPanelJsonView";
+	private static final String CAPTION_NAME = "Json";
+
+	private JTextArea httpPanelJsonArea;
+	private JPanel mainPanel;
+	private AbstractStringHttpPanelViewModel model;
+	private Pattern quickcheck;
+
+	public HttpPanelJsonView(AbstractStringHttpPanelViewModel model) {
+		httpPanelJsonArea = new HttpPanelJsonArea();
+		RTextScrollPane scrollPane = new RTextScrollPane(httpPanelJsonArea);
+		scrollPane.setLineNumbersEnabled(false);
+		mainPanel = new JPanel(new BorderLayout());
+		mainPanel.add(scrollPane, BorderLayout.CENTER);
+		this.model = model;
+		model.addHttpPanelViewModelListener(this);
+
+		//a very liberal regex for enabling the view!
+		quickcheck = Pattern.compile("\\A\\s*[{\\[\"].*[}\\]\"]\\s*\\z", Pattern.DOTALL);
+	}
+
+	@Override
+	public void setSelected(boolean selected) {
+		if (selected) {
+			httpPanelJsonArea.requestFocusInWindow();
+		}
+	}
+
+	@Override
+	public String getName() {
+		return NAME;
+	}
+
+	@Override
+	public String getCaptionName() {
+		return CAPTION_NAME;
+	}
+
+	@Override
+	public String getTargetViewName() {
+		return "";
+	}
+
+	@Override
+	public int getPosition() {
+		return 0;
+	}
+
+	@Override
+	public boolean isEnabled(Message message) {
+		if(message == null){
+			return false;
+		}
+		String jsonString = null;
+		if(message instanceof HttpMessage) {
+			HttpMessage httpMessage = ((HttpMessage) message);
+			if (this.model instanceof RequestBodyStringHttpPanelViewModel) {
+				jsonString = httpMessage.getRequestBody().toString();
+			} else if (this.model instanceof ResponseBodyStringHttpPanelViewModel) {
+				jsonString = httpMessage.getResponseBody().toString();
+			} else {
+				return false;
+			}
+		}
+		if(jsonString.length() > 1048576){
+			//parsing more than a mibibyte hangs ZAP
+			return false;
+		}
+		return quickcheck.matcher(jsonString).matches();
+	}
+
+	@Override
+	public boolean hasChanged() { return true; }
+
+	@Override
+	public JComponent getPane() {
+		return mainPanel;
+	}
+
+	@Override
+	public boolean isEditable() {
+		return httpPanelJsonArea.isEditable();
+	}
+
+	@Override
+	public void setEditable(boolean editable) {
+		httpPanelJsonArea.setEditable(editable);
+	}
+
+	@Override
+	public HttpPanelViewModel getModel() {
+		return model;
+	}
+
+	@Override
+	public void save() {
+		//we intentionally want to let the user give broken data
+		this.model.setData(httpPanelJsonArea.getText());
+	}
+
+	@Override
+	public void setParentConfigurationKey(String configurationKey) {
+	}
+
+	@Override
+	public void loadConfiguration(FileConfiguration fileConfiguration) {
+	}
+
+	@Override
+	public void saveConfiguration(FileConfiguration fileConfiguration) {
+	}
+
+	@Override
+	public void dataChanged(HttpPanelViewModelEvent e) {

```

Comment:
It should be called `httpPanelJsonArea.discardAllEdits();` after setting the value when not editable, otherwise it leaks memory.